datasciencebr/jarbas
# Use full text search to query reimbursements

9df9666 starts this WIP PR targeting #226

**What is already done**

* Adds `search_vector` field to `Reimbursement` model (with `None` as default value)
* Creates script to fill in the `search_vector` field (takes minutes/hours)

**How to test it**

1. First apply the new migrations<br>`$ python manage.py migrate`
1. Run the new search vector command<br>`$ python manage.py searchvector`
1. Play with the Django console:<br>`$ python manage.py shell_plus`

For example:

```python
from django.db.models import F
from django.contrib.postgres.search import SearchVector, SearchQuery, SearchRank

query = SearchQuery('buffet', config="portuguese")  # create a full text search query
rank = SearchRank(F('search_vector'), query)  # create a ranking based on this query

Reimbursement.objects.annotate(rank=rank) \
    .filter(search_vector=query) \
    .order_by('-rank')  # you should get a list of reimbursements with the world 'buffet'
```

**Next steps**

Adapt the dashboard search box to use this feature.
